### ISSUE TITLE
Exclude group from sharing after a share was created

### ISSUE THREAD

**Issue by PVince81** [owncloud/core#17346]
### Steps to reproduce

1. Login as admin
1. Create two users "user1" and "user2"
1. Add "user1" to group "group1"
1. Login as "user1"
1. Create a folder "test"
1. Upload 20-30 files into "test"
1. Scroll down and share any file on the second page (best is in the middle), share it with "user2"
1. Login as admin
1. Go to the admin page and set "Exclude groups from sharing" to "group1".
1. Login as "user2"

### Expected result
Shared file not visible.

### Actual result
Shared file is still there.
Additionally the JS code doesn't handle such situation very well, see https://github.com/owncloud/core/issues/15856#issuecomment-118049614

So basically there is a contradiction: the user still has an existing share but is now not allowed to share any more.

The problem is that "user1" has no way to remove that share. The only way is for "user2" to unshare from self.

This ticket is either about properly cleaning up shares after the share interdiction, or making sure that "forbidden" shares aren't visible any more.

@MorrisJobke @schiesbn @rullzer 

CC @jnfrmarks @jvillafanez for an interesting test case and inspiration for other similar ones

**Comment by rullzer** [comment]
Well this is really about what we expect the ```Exclude groups from sharing``` to mean.
I would expect that if a user is not able to share anything then old shares should be removed.

**Comment by rullzer** [comment]
First I was thinking just make the current shares invisible. So don't return them from the API. This is possible but does not scale. Since this would mean that each time we interact with the Sharing API we have to check from who the share we receive is (all the way up to the root share). As said this will not properly scale.

So the second option would be to have a hook somewhere. That if a group is added we check for all the users in that group if they now can't share anymore and if that is the case we remove all their shares.

I think the second option is cleanest. But it does risk removing all of a users shares if you accidentially enter the wrong group name...

**Comment by PVince81** [comment]
I don't think this has high priority though. Most important is that nothing breaks when this situation exists.

**Comment by PVince81** [comment]
not a bug then, by design. closing
